sections/featured-collection.liquid

```diff
@@ -78,7 +78,8 @@
         {%- for product in section.settings.collection.products limit: section.settings.products_to_show -%}
           <li
             id="Slide-{{ section.id }}-{{ forloop.index }}"
-            class="grid__item{% if show_mobile_slider or show_desktop_slider %} slider__slide{% endif %}"
+            class="grid__item{% if show_mobile_slider or show_desktop_slider %} slider__slide{% endif %}{% if settings.animations_reveal_on_scroll %}{% unless show_mobile_slider or show_desktop_slider %} scroll-trigger animate--slide-in{% endunless %}{% endif %}"
+            {% unless show_desktop_slider or show_mobile_slider %}data-cascade{% endunless %}
```

Comment:
I didn't see any dramatic doubling-up of the animation:

https://user-images.githubusercontent.com/22390758/228512701-1efdf956-dcb1-421c-a634-e8884d9057d3.mp4


Since we were already making use of `show_desktop_slider` and `show_mobile_slider`, I was trying to avoid duplicating these styles or having them in so many places. When we started doing animations, we always said we might have to tweak them at some point, and I think that relying on CSS too much will make it more difficult to do that, as we'll have to change things in lots of places, in a file that's already really large and difficult to navigate. Would it be a good idea to start moving these styles into their own file?

That being said, your suggestion made me realise there was an issue with my original code. The logic caused the cascading effect to stop working _on desktop_ when the mobile slider was enabled; and vice versa, cascading didn't work _on mobile_ when the desktop slider was enabled. Your suggestion fixes this 🎉

Reply:
My main concern with using liquid is just about getting the logic right. That issue you mention about the slider not working on desktop if the mobile slider is enabled is a good example. We don't know what type of client (desktop/mobile, touchscreen/mouse, safari/chrome/firefox, etc) it is in the liquid files, so we can't add or remove any markup to try and target certain devices. We can only use media queries in CSS and JS for that.

I think it's fine to use `animate--slide-in` heavily but in some cases you want to only apply the animation at certain breakpoints. An alternative to the solution above would be to keep a `animate--slide-in` class on the slider parent element but disable it with a more specific rule when there's not a carousel.

Looking forward for how we can do the cascading, I think it's a matter for a future PR :) But some ideas:
* Use JS with element.animate and media queries to do the animations for carousels (not CSS)
* Add a `<style>` block to the `featured-collection` liquid file so that you can inject theme settings into the CSS, and use `nth-child` rules to remove animations for slides that are out of view.